src/types/component.go

```diff
@@ -240,3 +241,16 @@ type ZarfComponentImport struct {
 	// For further explanation see https://regex101.com/r/nxX8vx/1
 	URL string `json:"url,omitempty" jsonschema:"description=[beta] The URL to a Zarf package to import via OCI,pattern=^oci://.*$"`
 }
+
+// JSONSchemaExtend extends the generated json schema during `zarf internal gen-config-schema`
+func (ZarfComponentImport) JSONSchemaExtend(schema *jsonschema.Schema) {
+	path, _ := schema.Properties.Get("path")
+	url, _ := schema.Properties.Get("url")
+
+	notSchema := &jsonschema.Schema{
+		Pattern: ZarfPackageTemplatePrefix,
+	}
+
+	path.Not = notSchema
+	url.Not = notSchema
+}
```

Comment:
Can we use the builtin `pattern=` struct tag to restrict URL and Path naming using regular expressions? Seems a bit redundant to have a pattern and a not pattern for the URL field, which also means we wouldn't need this `JSONSchemaExtend` function

Reply:
I could be wrong here, but from what I've seen this is the only way as doing as a normal pattern would require a negative lookahead which is not allowed in go regex.